From 08b94155141151123737ff72a8cdd9c818de2e68 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20=C5=BBo=C5=82nowski?= <aidecoe@aidecoe.name>
Date: Sat, 26 Feb 2011 08:08:37 +0100
Subject: [PATCH 2/2] Gentoo ebuild for LVM2 prior to 2.02.63-r1 doesn't install rules:

  10-dm.rules
  11-dm-lvm.rules
  13-dm-disk.rules
  95-dm-notify.rules

but provides only 64-device-mapper.rules combining more or less the
above rules files.
---
 modules.d/90dm/module-setup.sh     |    3 +++
 modules.d/90dmraid/module-setup.sh |    3 +++
 modules.d/90lvm/module-setup.sh    |    3 +++
 3 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/modules.d/90dm/module-setup.sh b/modules.d/90dm/module-setup.sh
index 11985fd..67b4d40 100755
--- a/modules.d/90dm/module-setup.sh
+++ b/modules.d/90dm/module-setup.sh
@@ -22,5 +22,8 @@ install() {
 
     type -P dmeventd >/dev/null && dracut_install dmeventd
     inst_rules 10-dm.rules 95-dm-notify.rules
+    # Gentoo ebuild for LVM2 prior to 2.02.63-r1 doesn't install above rules
+    # files, but provides the one below:
+    inst_rules 64-device-mapper.rules
 }
 
diff --git a/modules.d/90dmraid/module-setup.sh b/modules.d/90dmraid/module-setup.sh
index f437712..6950d3c 100755
--- a/modules.d/90dmraid/module-setup.sh
+++ b/modules.d/90dmraid/module-setup.sh
@@ -43,6 +43,9 @@ install() {
     done
 
     inst_rules 10-dm.rules 13-dm-disk.rules 95-dm-notify.rules 
+    # Gentoo ebuild for LVM2 prior to 2.02.63-r1 doesn't install above rules
+    # files, but provides the one below:
+    inst_rules 64-device-mapper.rules
 
     inst "$moddir/dmraid.sh" /sbin/dmraid_scan
 
diff --git a/modules.d/90lvm/module-setup.sh b/modules.d/90lvm/module-setup.sh
index 2555728..f3da242 100755
--- a/modules.d/90lvm/module-setup.sh
+++ b/modules.d/90lvm/module-setup.sh
@@ -47,6 +47,9 @@ install() {
     fi
 
     inst_rules 10-dm.rules 13-dm-disk.rules 95-dm-notify.rules 11-dm-lvm.rules
+    # Gentoo ebuild for LVM2 prior to 2.02.63-r1 doesn't install above rules
+    # files, but provides the one below:
+    inst_rules 64-device-mapper.rules
 
     inst "$moddir/lvm_scan.sh" /sbin/lvm_scan
     inst_hook cmdline 30 "$moddir/parse-lvm.sh"
-- 
1.7.4.1

